src/main/java/seedu/address/model/task/repeatable/Event.java

```diff
@@ -82,6 +77,10 @@ public int hashCode() {
      */
     @Override
     public String toString() {
-        return this.description + " (at: " + DateUtil.decodeDate(at) + ")";
+        if (getIsWeekly() == false) {
+            return this.description + " (at: " + DateUtil.decodeDate(date) + " " + TimeUtil.decodeTime(time) + ")";
+        }
+
+        return this.description + " (at: " + DateUtil.decodeDate(date) + " " + TimeUtil.decodeTime(time) + " [WEEKLY])";
```

Comment:
If it's weekly maybe better to show the day instead of the date.

Reply:
thanks!